src/components/ImportOnyxState/index.native.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import RNFS from 'react-native-fs';
+import Onyx from 'react-native-onyx';
+import type {FileObject} from '@components/AttachmentModal';
+import AttachmentPicker from '@components/AttachmentPicker';
+import * as Expensicons from '@components/Icon/Expensicons';
+import MenuItem from '@components/MenuItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {KEYS_TO_PRESERVE, setIsUsingImportedState} from '@libs/actions/App';
+import {setShouldForceOffline} from '@libs/actions/Network';
+import Navigation from '@libs/Navigation/Navigation';
+import type {OnyxValues} from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type ImportOnyxStateProps from './types';
+import {cleanAndTransformState} from './utils';
+
+const CHUNK_SIZE = 100;
+
+function readFileInChunks(fileUri: string, chunkSize = 1024 * 1024) {
+    const filePath = decodeURIComponent(fileUri.replace('file://', ''));
+
+    return RNFS.exists(filePath)
+        .then((exists) => {
+            if (!exists) {
+                throw new Error('File does not exist');
+            }
+            return RNFS.stat(filePath);
+        })
+        .then((fileStats) => {
+            const fileSize = fileStats.size;
+            let fileContent = '';
+            const promises = [];
+
+            // Chunk the file into smaller parts to avoid memory issues
+            for (let i = 0; i < fileSize; i += chunkSize) {
+                promises.push(RNFS.read(filePath, chunkSize, i, 'utf8').then((chunk) => chunk));
+            }
+
+            // After all chunks have been read, join them together
+            return Promise.all(promises).then((chunks) => {
+                fileContent = chunks.join('');
+
+                return fileContent;
+            });
+        })
+        .catch((error) => {
+            throw error;
+        });
+}
+
+function chunkArray<T>(array: T[], size: number): T[][] {
+    const result = [];
+    for (let i = 0; i < array.length; i += size) {
+        result.push(array.slice(i, i + size));
+    }
+    return result;
+}
+
+function applyStateInChunks(state: OnyxValues) {
+    const entries = Object.entries(state);
+    const chunks = chunkArray(entries, CHUNK_SIZE);
+
+    let promise = Promise.resolve();
+    chunks.forEach((chunk) => {
+        const partialOnyxState = Object.fromEntries(chunk) as Partial<OnyxValues>;
+        promise = promise.then(() => Onyx.multiSet(partialOnyxState));
+    });
+
+    return promise;
```

Comment:
Correct me if I am wrong but it looks like we are overriding the promise from previous chunk with a promise from next chunk. Is this desired behaviour?

What happens with promises generated from chunks before the very last one?

Reply:
it creates a new promise that waits for the previous promise to resolve before applying the next chunk, creating a chain of promises. This ensures that the `Onyx.multiSet` operation for each chunk will execute in sequence, waiting for the previous operation to complete

Reply:
Oh right, that's nice!

Thanks for explanation ❤️